flink-connectors/flink-connector-pulsar/src/main/java/org/apache/flink/connector/pulsar/source/enumerator/PulsarSourceEnumStateSerializer.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.connector.pulsar.source.enumerator;
+
+import org.apache.flink.connector.pulsar.source.enumerator.topic.TopicPartition;
+import org.apache.flink.connector.pulsar.source.split.PulsarPartitionSplit;
+import org.apache.flink.connector.pulsar.source.split.PulsarPartitionSplitSerializer;
+import org.apache.flink.core.io.SimpleVersionedSerializer;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.DataInput;
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.util.Map;
+import java.util.Set;
+
+import static org.apache.flink.connector.pulsar.common.utils.PulsarSerdeUtils.deserializeMap;
+import static org.apache.flink.connector.pulsar.common.utils.PulsarSerdeUtils.deserializeSet;
+import static org.apache.flink.connector.pulsar.common.utils.PulsarSerdeUtils.serializeMap;
+import static org.apache.flink.connector.pulsar.common.utils.PulsarSerdeUtils.serializeSet;
+
+/** The {@link SimpleVersionedSerializer Serializer} for the enumerator state of Pulsar source. */
+public class PulsarSourceEnumStateSerializer
+        implements SimpleVersionedSerializer<PulsarSourceEnumState> {
+
+    private static final PulsarPartitionSplitSerializer SPLIT_SERIALIZER =
+            new PulsarPartitionSplitSerializer();
```

Comment:
Maybe use singleton pattern in `PulsarPartitionSplitSerializer` and just fetch instance here.
This `PulsarSourceEnumStateSerializer` could also be a singleton.